lib/contextual_logger/redactor.rb

```diff
@@ -17,11 +17,18 @@ def register_secret(sensitive_data)
       end
     end
 
-    def redact(log_line)
+    def redact(log_entry)
       if redaction_regex
-        log_line.gsub(redaction_regex, '<redacted>')
+        case log_entry
+        when Hash
+          log_entry.reduce({}) { |redacted_log_entry, (key, value)| redacted_log_entry.merge(key => redact(value)) }
+        when Array
```

Comment:
The values of the context hash can be anything, and this gem doesn't confine to a specific level of the context hash, it can be anything that's necessary for the added context, and arrays are definitely useful and usable when importing context to Elasticsearch.  This gem at it's base just `to_json`'s the message with context, so `false` stays as `false`, it doesn't get stringified.  Which that is a good reminder and I've gone ahead and removed the `to_s` case for `true` and `false` to preserve them as booleans in the end context.

Confinement to a specific context level should be appended to the formatting using the log formatter rather than enforced by this gem.

Reply:
Ok, I think I was confusing the false bit with what happens to the `message` param. *That* gets to_s'd.